### DISCUSSION TITLE
How to GET the list of dependabot alerts via GitHub API?

### DISCUSSION THREAD

**Discussion by rodrigobercini** [community/community#24309] [Apps, API and Webhooks]
<p>How can I GET the list of dependabot alerts available at <a href="https://github.com/%7Buser%7D/%7Brepo%7D/security/dependabot?page=1&amp;q=is%3Aopen" rel="noopener nofollow ugc">https://github.com/{user}/{repo}/security/dependabot?page=1&amp;q=is%3Aopen</a> via the GitHub API?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/0/70b04a7b1133b9c2895a026c65138978c02d66d2.png" data-download-href="/uploads/short-url/g4Tb3l2nanv9ZG4SVD15kxSDegO.png?dl=1" title="dependabot alerts" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/18664907/181084182-422503b6-4d86-44af-b012-c35e142fab67.png" alt="dependabot alerts" data-base62-sha1="g4Tb3l2nanv9ZG4SVD15kxSDegO" width="690" height="191" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/0/70b04a7b1133b9c2895a026c65138978c02d66d2_2_690x191.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/0/70b04a7b1133b9c2895a026c65138978c02d66d2_2_1035x286.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/7/0/70b04a7b1133b9c2895a026c65138978c02d66d2_2_1380x382.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/0/70b04a7b1133b9c2895a026c65138978c02d66d2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">dependabot alerts</span><span class="informations">1628×453 21.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I searched through <a href="https://docs.github.com/en/rest" rel="noopener nofollow ugc">the documentation</a> but couldn’t find anything there.</p>
<p>Thanks!</p>

**Comment by rodrigobercini** [comment, marked as answer]
<p>Found in the GraphQL API <a href="https://docs.github.com/en/graphql/reference/objects#repositoryvulnerabilityalert" rel="noopener nofollow ugc">RepositoryVulnerabilityAlert</a>:</p>
<pre><code class="lang-auto">{
    repository(name: "repo-name", owner: "repo-owner") {
        vulnerabilityAlerts(first: 100) {
            nodes {
                createdAt
                dismissedAt
                securityVulnerability {
                    package {
                        name
                    }
                    advisory {
                        description
                    }
                }
            }
        }
    }
}
</code></pre>
<p>Thanks to <a href="https://stackoverflow.com/users/2614364/bertrand-martel" class="inline-onebox" rel="noopener nofollow ugc">User Bertrand Martel - Stack Overflow</a></p>

**Reply by slice-rishabhd** [reply]
This gives list of first 100 alerts only for a repo, what if there are more than that, how to list them all ??

EDIT : found the answer to my question, you can just use this for the same : 
```
query ($orgName: String!,$cursor : String,$repoName: String!) {
  repository(name: $repoName, owner: $orgName) {
    vulnerabilityAlerts(first: 100, after: $cursor) {
      pageInfo {
        hasNextPage
        endCursor
      }
      nodes {
        createdAt
        dismissedAt
        securityVulnerability {
          package {
            name
          }
          advisory {
            description
          }
        }
      }
    }
  }
}
``` 

Just keep updating the cursor variable from the reponse until hasNextPage becomes null and you will get all the alerts in a repo.
